.github/workflows/ci.yml

```diff
@@ -34,52 +34,72 @@ jobs:
           pip install wheel
           curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
 
+      - name: Setup cache variables
+        shell: bash
+        id: cache-vars
+        run: |
+          export PATH="$HOME/.poetry/bin:$PATH"
+          echo "::set-output name=poetry-dir::$(poetry config cache-dir)"
+          echo "::set-output name=date::$(/bin/date -u "+%m%w%Y")"
+
+      - name: Setup Poetry cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.cache-vars.outputs.poetry-dir }}
+          key: ${{ runner.os }}-poetry-${{ matrix.python }}-${{ hashFiles('poetry.lock') }}
+
+      - name: Install and cache ffmpeg (all OS)
+        uses: FedericoCarboni/setup-ffmpeg@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+        id: setup-ffmpeg
+
       - name: Install system dependencies (Linux)
         if: runner.os == 'Linux'
         run: |
           sudo apt update
-          sudo apt install -y ffmpeg
           sudo apt-get -y install texlive texlive-latex-extra texlive-fonts-extra texlive-latex-recommended texlive-science texlive-fonts-extra tipa python-opengl libpango1.0-dev xvfb
           echo "$HOME/.poetry/bin" >> $GITHUB_PATH
           # start xvfb in the background
           sudo /usr/bin/Xvfb $DISPLAY -screen 0 1280x1024x24 &
 
+      - name: Setup Brew cache (MacOS)
+        id: cache-brew
+        if: runner.os == 'macOS'
+        uses: actions/cache@v2
+        with:
+          path: ~/brew-cache
+          # Will become ${{ runner.os }}-brew-${{ matrix.python }}-${{ steps.cache-vars.outputs.date }}
+          # after cache implementation is complete
+          key: test8-brew-${{ matrix.python }}-${{ steps.cache-vars.outputs.date }}
+
       - name: Install system dependencies (MacOS)
         if: runner.os == 'macOS'
         run: |
-          brew update && brew upgrade basictex && brew cleanup
-          brew install openssl readline ffmpeg pyenv pyenv-virtualenv
+          brew update -q && brew upgrade basictex && brew cleanup
```

Comment:
we should start using tinytex...

Reply:
Yea that's a good idea.

Reply:
Pip and poetry are done, so I'll work on tinytex soon.